File: core/src/main/java/com/simple/castle/core/ServerState.java
package com.simple.castle.core;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public final class ServerState implements Serializable {
    private final List<ModelSend> modelSends;

    public ServerState(List<ModelSend> modelSends) {
        this.modelSends = new ArrayList<>();

        for (ModelSend modelSend : modelSends) {
            this.modelSends.add(new ModelSend(modelSend.getId(), modelSend.getMatrix4()));
        }
    }

    public List<ModelSend> getModelSends() {
        return modelSends;
    }

    @Override
    public String toString() {
        return "ServerState{" +
                "modelSends=" + modelSends +
                '}';
    }
}

File: core/src/main/java/com/simple/castle/core/kt/api/ServerState.kt
package com.simple.castle.core.kt.api

import java.io.Serializable
import java.util.*

class ServerState(modelSends: List<ModelSend>) : Serializable {
    private val modelSends: MutableList<ModelSend>
    fun getModelSends(): List<ModelSend> {
        return modelSends
    }

    override fun toString(): String {
        return "ServerState{" +
                "modelSends=" + modelSends +
                '}'
    }

    init {
        this.modelSends = ArrayList()
        for (modelSend in modelSends) {
            this.modelSends.add(ModelSend(modelSend.id, modelSend.matrix4))
        }
    }
}